public/components/FeatureAnywhereContextMenu/AssociatedDetectors/containers/AssociatedDetectors.tsx

```diff
@@ -0,0 +1,347 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useMemo, useEffect, useState } from 'react';
+import {
+  EuiFlyoutHeader,
+  EuiTitle,
+  EuiSpacer,
+  EuiInMemoryTable,
+  EuiFlyoutBody,
+  EuiButton,
+  EuiFlyout,
+  EuiFlexItem,
+  EuiFlexGroup,
+} from '@elastic/eui';
+import { get, isEmpty } from 'lodash';
+import '../styles.scss';
+import { getColumns } from '../utils/helpers';
+import { useDispatch, useSelector } from 'react-redux';
+import { AppState } from '../../../../redux/reducers';
+import { DetectorListItem } from '../../../../models/interfaces';
+import { getSavedFeatureAnywhereLoader } from '../../../../services';
+import {
+  GET_ALL_DETECTORS_QUERY_PARAMS,
+  SINGLE_DETECTOR_NOT_FOUND_MSG,
+} from '../../../../pages/utils/constants';
+import { getDetectorList } from '../../../../redux/reducers/ad';
+import {
+  prettifyErrorMessage,
+  NO_PERMISSIONS_KEY_WORD,
+} from '../../../../../server/utils/helpers';
+import { SavedObjectLoader } from '../../../../../../../src/plugins/saved_objects/public';
+import { EmptyAssociatedDetectorFlyoutMessage } from '../components/EmptyMessage/EmptyMessage';
+import { ISavedAugmentVis } from '../../../../../../../src/plugins/vis_augmenter/public';
+import { ASSOCIATED_DETECTOR_ACTION } from '../utils/constants';
+import { ConfirmUnlinkDetectorModal } from '../components/ConfirmUnlinkDetectorModal/ConfirmUnlinkDetectorModal';
```

Comment:
nit - you can add `index.ts` files for each directory and sub-directory to handle the exports. This way it simplifies imports since you just have to reference directories instead of the implemented file. e.g., changing the above to 
```
import { ConfirmUnlinkDetectorModal } from '../components';
```

Reply:
I added a few index.ts to handle exports, should we do this even for every components folder that just has one file? didn't want to cause way too many files.

Reply:
Added this